### DISCUSSION TITLE
Anything Wrong with my codes? Contact sensor didnt feedback..

### DISCUSSION THREAD

**Discussion by dylanfly99** [HomeSpan/HomeSpan#714] [Q&A]
Ask for help...I am newest to lean Arduino for ESP32. I love this HOMESPAN.
When I connect it to my HOME. And touch the reedSwitch then the  Door ICON in my HOME did not respond.
Here is one of my part of codes:
const int reedSwitchPin = 23; 

struct DEV_ContactSensor : Service::ContactSensor {
  SpanCharacteristic *contactSensorState;

   DEV_ContactSensor() : Service::ContactSensor() {
    pinMode(reedSwitchPin, INPUT_PULLUP); 
    contactSensorState = new Characteristic::ContactSensorState(true); // 
  }

  boolean update() {
    
    if (digitalRead(reedSwitchPin) == HIGH) {
      contactSensorState->setVal(true); 
    } else {
      contactSensorState->setVal(false); 
    }
    return (true);
  }
};

**Comment by HomeSpan** [comment]
First, check to make sure the sensor is recording what you think it's supposed to do.  The best way too do that is to add some Serial.print statements to your logic above (something like `Serial.print("Reed Switch Activated");`)

My guess is that the reed switch may be noisy, which means it will send a lot updates.  If that is the issue, use SpanButton (see the API and examples for details) for the reed switch.
